lib/event_store/storage/appender.ex

```diff
@@ -117,10 +108,24 @@ defmodule EventStore.Storage.Appender do
     event_id |> uuid()
   end
 
-  defp insert_event_batch(conn, events, schema, opts) do
-    event_count = length(events)
-    statement = Statements.insert_events(schema, event_count)
-    parameters = build_insert_parameters(events)
+  defp insert_event_batch(
+         conn,
+         stream_id,
+         stream_uuid,
+         events,
+         event_count,
+         schema,
+         any_version,
+         opts
+       ) do
+    statement =
+      case any_version do
+        true -> Statements.insert_events_any_version(schema, stream_id, event_count)
+        false -> Statements.insert_events(schema, stream_id, event_count)
+      end
```

Comment:
Can these two SQL statements be combined into one, possibly by passing the `any_version` as an additional argument to the function? Otherwise needing to keep both scripts updated and exhibiting identical behaviour for future changes is a concern.

Reply:
I'm worried that having a bunch of `if` statements in these queries will lead to them being more difficult to decipher. Your SQL-fu has to already be pretty good to make sense of these queries. The structure is similar in some ways, but in other ways it is not.

I'm not sure if combining these two SQL queries with if statements will change the fundamentals of the test situation.

Reply:
Previously appending events to a stream used the same SQL statements regardless of which concurrency checking was used (`:any_version`, `:no_stream`, `:stream_exists`, or expected version). Most of the tests are using the expected version (e.g. appending events to a stream ensures they can be read from "all events"). By splitting the two into separate SQL statements we'd need to increase the test coverage for the `:any_version` to cover all the expected behaviour.

Reply:
I have time, but I am just not sure how to make one SQL statement that does both. We don't fundamentally change the testing situation by leaving the SQL the same but adding `if` statements.

Perhaps it makes sense to add at least a test to check that `:any_version` links events into the global `$all` stream.

I'll hit you up on slack to discuss.

Reply:
> We don't fundamentally change the testing situation by leaving the SQL the same but adding if statements.

If the only difference between the two approaches is the SQL to link the events into the source stream, and the SQL to link events to the global `$all` stream is shared, then tests for one approach should also be valid for the others.